### PR TITLE
Fix #498 by adding Google Cloud Functions support

### LINKED ISSUE #498: Add bolt-google-cloud-functions (Google Cloud Functions for Java 11)
### Issue Type

- [ ] Bug
- [x] Enhancement / Feature request
- [ ] Question
- [ ] Documentation

### Description

Google Cloud Functions started supporting Java 11 apps.
https://cloud.google.com/blog/products/application-development/introducing-java-11-on-google-cloud-functions

As with [bolt-aws-lambda](https://github.com/slackapi/java-slack-sdk/tree/master/bolt-aws-lambda), we can provide an optional module for the cloud runtime users.

### Requirements (place an `x` in each of the `[ ]`)

* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by seratch**
This pull request adds a new module `bolt-google-cloud-functions` that enables developers to easily implement Google Cloud's `HttpFunction`. Fixes #498 

```java
package functions;
import com.slack.api.bolt.App;

public class HelloSlack extends SlackApiFunction {
    private static final App app = new App();

    static {
        app.command("/hi", (req, ctx) -> {
            return ctx.ack("Hi from Google Cloud Functions!");
        });
    }

    public HelloSlack() {
        super(app);
    }
}
```

```bash
export SLACK_BOT_TOKEN=
export SLACK_SIGNING_SECRET=

// Local Development
mvn function:run
ngrok http 8080 --subdomain your-domain

// Deployment
gcloud functions deploy my-first-function --entry-point functions.HelloSlack --runtime java11 --trigger-http --memory 512MB --allow-unauthenticated --set-env-vars SLACK_BOT_TOKEN=$SLACK_BOT_TOKEN,SLACK_SIGNING_SECRET=$SLACK_SIGNING_SECRET
```

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [x] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

File: bolt-google-cloud-functions-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.slack.api.samples</groupId>
+    <artifactId>bolt-google-cloud-functions-example</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <properties>
+        <google-cloud-functions.version>1.0.2</google-cloud-functions.version>
+        <bolt.version>[1.1,)</bolt.version>
+        <maven.compiler.target>11</maven.compiler.target>
```
Comment (seratch) [inline bolt-google-cloud-functions-example/pom.xml:12]: This setting overwrites the compatible Java version (the default is Java 8)

File: bolt-google-cloud-functions-example/src/main/java/functions/HelloSlack.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// Deployment
+// gcloud functions deploy my-first-function --entry-point functions.HelloSlack --runtime java11 --trigger-http --memory 512MB --allow-unauthenticated --set-env-vars SLACK_BOT_TOKEN=$SLACK_BOT_TOKEN,SLACK_SIGNING_SECRET=$SLACK_SIGNING_SECRET
+
+/**
+ * Refer to https://cloud.google.com/functions/docs/first-java for details.
+ */
+public class HelloSlack extends SlackApiFunction {
+    private static final App app = new App();
+
+    static {
```
Comment (seratch) [inline bolt-google-cloud-functions-example/src/main/java/functions/HelloSlack.java:21]: You don't need to embed the logic here but this is the handiest way.

File: bolt-google-cloud-functions/pom.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (seratch) [inline bolt-google-cloud-functions/pom.xml:1]: This pom.xml is based on https://cloud.google.com/functions/docs/first-java This Maven project is not part of the multi-module maven project. It works independently (this means it downloads jars from the Maven Central and boots the app without any extra settings).

File: pom.xml [hunk 4]
```diff
@@ -35,6 +36,8 @@
     </parent>
 
     <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (seratch) [inline pom.xml:39]: These settings enforce the projects in this repository to build jars that are compatible with Java 8 by default. Some projects (bolt-google-cloud-functions, bolt-helidon in the short term) may overwrite the version.

Maintainers must use OpenJDK 11 (or newer stable LTS version in the future) for building jars from now on.

**Review by seratch — COMMENTED** [review]
Comments for reviewers

**Comment by seratch** [comment]
Once we ship this optional module, we will be updating README and the top page of the website.

**Comment by seratch** [comment]
We will merge this after v1.1.4 release.

File: bolt-google-cloud-functions-example/src/main/java/functions/SlackApiFunction.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ * An HttpFunction that handles incoming requests from Slack.
+ */
+public class SlackApiFunction implements HttpFunction {
```
Comment (seratch) [inline bolt-google-cloud-functions-example/src/main/java/functions/SlackApiFunction.java:21]: will remove this after releasing the initial version.

File: scripts/release.sh [hunk 6]
```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
 
+is_jdk_8=`echo $JAVA_HOME | grep 8.`
+is_travis_jdk_8=`echo $TRAVIS_JDK | grep openjdk8`
+if [[ "${is_jdk_8}" != "" && "${is_travis_jdk_8}" != "" ]];
+then
+  echo "Please use OpenJDK 11 for releasing these libraries."
+  exit 1
+fi
+
```
Comment (mwbrooks) [inline scripts/release.sh:10]: Nice validation check in the release script!

**Review by mwbrooks — APPROVED** [review]
Looks good to me! It's great to see Google Cloud Function support landing.

**Pull request merged**